src/garage/torch/value_functions/base.py

```diff
@@ -0,0 +1,71 @@
+"""Base class for all baselines."""
+import abc
+
+
+class ValueFunction(abc.ABC):
+    """Base class for all baselines.
+
+    Args:
+        env_spec (garage.envs.env_spec.EnvSpec): Environment specification.
+        name (str): Value function name, also the variable scope.
+
+    """
+
+    def __init__(self, env_spec, name):
+        self._mdp_spec = env_spec
+        self.name = name
+
+    @abc.abstractmethod
+    def get_param_values(self):
+        """Get parameter values.
+
+        Returns:
+            List[np.ndarray]: A list of values of each parameter.
+
+        """
+
+    @abc.abstractmethod
+    def set_param_values(self, flattened_params):
+        """Set param values.
+
+        Args:
+            flattened_params (np.ndarray): A numpy array of parameter values.
+
+        """
+
+    @abc.abstractmethod
+    def compute_loss(self, obs, returns):
+        r"""Compute mean value of loss.
+
+        Args:
+            obs (torch.Tensor): Observation from the environment
+                with shape :math:`(N \dot [T], O*)`.
+            returns (torch.Tensor): Acquired returns with shape :math:`(N, )`.
+
+        Returns:
+            torch.Tensor: Calculated negative mean scalar value of
+                objective (float).
+
+        """
+
+    @abc.abstractmethod
+    def predict(self, obs):
```

Comment:
agreed. the idea in this PR is to just consider the VF as any other NN primitive in garage. in torch, that means you retrieve its values with forward. this will allow us to compute the policy loss with a single feedforward, for instance. it also allows us to apply things like MAML to the VF as well as the policy.

Reply:
Every value function returns distribution object?
If I remove the `predict` function, I need to call `self._value_function(obs).mean.flatten(02)`
but what if there is a value function that directly returns baseline value instead of distribution object?

Reply:
fixed